primitives/src/traits/dispute_api.rs

```diff
@@ -51,4 +51,62 @@ pub trait DisputeApi {
         market_id: &Self::MarketId,
         market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
     ) -> Result<Option<OutcomeReport>, DispatchError>;
+
+    /// Query the future resolution block of a disputed market.
+    /// Fails if the market dispute mechanism does not belong to the actual dispute mechanism.
+    ///
+    /// # Returns
+    ///
+    /// Returns the future resolution block if available, otherwise `None`.
+    fn get_auto_resolve(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<Option<Self::BlockNumber>, DispatchError>;
+
+    /// Query if the dispute mechanism failed for a dispute market.
+    /// Fails if the market dispute mechanism does not belong to the actual dispute mechanism.
+    ///
+    /// # Returns
+    ///
+    /// Returns true, when the dispute mechanism failed. Otherwise false.
+    fn is_fail(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<bool, DispatchError>;
+}
+
+pub trait DisputeResolutionApi {
+    type AccountId;
+    type BlockNumber;
+    type MarketId;
+    type Moment;
+
+    /// Resolve a market. Fails if `on_resolution` from zrml-prediction-markets fails.
+    ///
+    /// **Should only be called if the market dispute**
+    /// **mechanism is ready for the resolution (`DisputeApi::on_resolution`).**
+    ///
+    /// # Returns
+    ///
+    /// Returns the consumed weight.
+    fn resolve(
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<u64, DispatchError>;
```

Comment:
> What shall we do? Using Weight also in prediction-markets on_resolution?

Seems reasonable. The fact that `on_resolution` uses `u64` looks like an oversight rather than an intended choice.

Reply:
965478095708d07d99e9b67ff54091291abcc953